src/constructors/lattices.c

```diff
@@ -288,20 +296,303 @@ igraph_error_t igraph_triangular_lattice(
+        VECTOR(row_lengths_vector)[i] = row_length;
+        VECTOR(row_start_vector)[i] = row_start;
+
+        if (i < first_threshold) {
+            row_length += 2;
+            row_start -= 2;
+        } else if (i < second_threshold) {
+            row_start += sgn_flag;
+        } else {
+            row_length -= 2;
+        }
+        if (i == size_y - 1) {
+            row_start--;
+            row_length++;
+        }
+        if (i == size_z - 1) {
+            row_length++;
+        }
+    }
+
+    IGRAPH_CHECK(hexagonal_lattice(graph, directed, mutual, &row_lengths_vector, &row_start_vector));
+
+    igraph_vector_int_destroy(&row_lengths_vector);
+    igraph_vector_int_destroy(&row_start_vector);
+    IGRAPH_FINALLY_CLEAN(2);
+
+    return IGRAPH_SUCCESS;
+}
+
+/**
+ * \function igraph_hexagonal_lattice
+ * \brief A hexagonal lattice with the given shape.
+ *
+ * \experimental
+ *
+ * Creates a hexagonal lattice whose vertices have the form (i, j) for non-negative integers i and j
+ * and (i, j) is generally connected with (i + 1, j), and if i is odd also with (i - 1, j + 1).
+ * The function constructs a planar dual of the graph constructed by <code>igraph_triangular_lattice</code>.
+ * In particular, there a one-to-one correspondence between the cycles of length 6 in the constructed graph
+ * and the vertices of the graph constructed by <code>igraph_triangular_lattice</code> function
+ * with the same \p dims parameter.
+ *
+ * </para><para>
+ * The vertices of the resulting graph are ordered lexicographically with the 2nd coordinate being
+ * more significant, e.g., (i, j) &lt; (i + 1, j) and (i + 1, j) &lt; (i, j + 1)
+ *
+ * \param graph An uninitialized graph object.
+ * \param dims Integer vector, defines the shape of the lattice. (Below the "edge length"s are in terms of graph theoretical path lengths.)
+ *        If \p dims is of length 1, the resulting lattice has a triangular shape
+ *        where each side of the triangle contains <code>dims[0]</code> vertices.
+ *        If \p dims is of length 2, the resulting lattice has a
+ *        "quasi rectangular" shape with the sides containing <code>dims[0]</code> and
+ *        <code>dims[1]</code> vertices, respectively.
+ *        If \p dims is of length 3, the resulting lattice has a hexagonal shape
+ *        where the sides of the hexagon contain <code>dims[0]</code>, <code>dims[1]</code> and
+ *        <code>dims[2]</code> vertices.
+ *        All coordinates must be non-negative.
+ * \param directed Boolean, whether to create a directed graph.
+ *        If the \c mutual argument is not set to true,
+ *        edges will be directed from lower-index vertices towards
+ *        higher-index ones.
+ * \param mutual Boolean, if the graph is directed this gives whether
+ *        to create all connections as mutual.
+ * \return Error code:
+ *         \c IGRAPH_EINVAL: The size of \p dims must be either 1, 2, or 3 with all the components
+ *         at least 1.
+ *
```

Comment:
```suggestion
 * \sa \ref igraph_triangular_lattice() for creating a triangular lattice.
 *
```

A similar `\sa` reference should be added in the docs of `igraph_triangular_lattice()` as well.